src/common/constants.ts

```diff
@@ -3,46 +3,62 @@ export const constants = {
     strategy: 'jwt',
   },
   database: 'database',
-  errors: {
-    schema_validation: {
-      name: 'SCHEMA_VALIDATION',
-      message: 'One or more fields contain incorrect values. Please verify and try again.',
-      code: 1000,
-    },
-    unauthorized: {
-      default: {
-        name: 'UNAUTHORIZED',
-        message: 'Unauthorized. You do not have permission to complete your request.',
-        code: 2000,
-      },
-    },
-    not_found: {
-      default: {
-        name: 'NOT_FOUND',
-        message: 'Not found. The resource that you are looking for does not exist.',
-        code: 3000,
-      },
-    },
-    business_exception: {
-      default: {
-        name: 'BUSINESS_RULES',
-        message:
-          'An error has occurred while processing your request. Please verify your information and try again.',
-        code: 4000,
-      },
-      address_not_associated: {
-        name: 'BUSINESS_RULES',
-        message: 'The given address does not have an associated account.',
-        code: 4001,
-      },
-    },
-    internal_exception: {
-      default: {
-        name: 'INTERNAL',
-        message:
-          'An internal error has occurred. Please help us improve our service by sending an error report.',
-        code: 5000,
-      },
-    },
+};
+
+export const SchemaValidationErrors = {
+  default: {
+    name: 'SCHEMA_VALIDATION',
+    message: 'One or more fields contain incorrect values. Please verify and try again.',
+    code: 1000,
+  },
+};
+
+export const UnauthorizedErrors = {
+  default: {
+    name: 'UNAUTHORIZED',
+    message: 'Unauthorized. You do not have permission to complete your request.',
+    code: 2000,
+  },
+};
+
+export const NotFoundErrors = {
```

Comment:
Might be a little bit confusing say "**NotFoundErrors**".
I'd say "**ResourceNotFoundErrors**" or something like that instead.

Reply:
hmm, good point, I've been thinking about these constants. It might be a good idea to merge unauthorized, validation and notfound errors together and make only one set of errors, so we'll end up with three kinds, http error, business and internal.. what do you think?

Reply:
Yeah, I think that's a good idea, so we could handle easier the errors, and if we are catching up one, we just call the object and show the respective error.
i.e: user enters to a page that doesn't exist, show an error from http error, like 404 with a msg "Page not found", or something like that. 
something like: `throw new httpError.resourceNotFound`

Reply:
The question is, Nest already has an exception for this behavior, so, I think so we just need constants for the messages

Reply:
Yeah, agree, nest already have exception handling OOB. However, what we are looking for with this solution is  composing a custom error response, which contains more information, not only a statusCode and a name, i.e. error code, path, timestamp, name, description. Hence we had to create an 'enhanced' version of it using a global exception filter. The idea behind the constants though, is to make the exception throwing easier.